feat: Removing old attributes and changing the identity attributes used

markvdouw opened the pull request:
## Instructions
 1. PR target branch should be against `development`
 2. PR title name should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-title-check.yml
 3. PR branch prefix should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-branch-check-name.yml

 ## Summary
 - Removing old attributes and changing the identity function to register base on new attributes:
 "userIdentificationType": "CustomerId"
 "emailIdentificationType": "Email"
Adding catch block to handle issue with deeplinks preventing core-sdk to finish initilization of the kit.
Implementing IdentityListener

E2E testing (checking unique user in the day after changing values)
<img width="1448" alt="Screen Shot 2023-09-07 at 11 53 10" src="https://github.com/mparticle-integrations/mparticle-android-integration-kochava/assets/9582473/f18b4b50-937b-4503-be4b-ea920899c8eb">


 ## Testing Plan
 - [ ] Was this tested locally? If not, explain why. - Need valid kochava account
 - {explain how this has been tested, and what, if any, additional testing should be done}

 ## Reference Issue (For mParticle employees only.  Ignore if you are an outside contributor)
 - Closes https://mparticle-eng.atlassian.net/browse/SQDSDKS-5726

Local and E2E test pending on valid Kochava account.

Commit:
Removing old attributes and changing the identity function to register base on new attributes:
 "userIdentificationType": "CustomerId"
 "emailIdentificationType": "Email"

BrandonStalnaker reviewed (commented):
The code itself LGTM but does the Android code ever check existing identities and send them? For example, if a user has already logged in and had all their identities set before customer adds Kochava integration (or this update) this code will never send identity for them.

Commit:
Add IdentityListener since the last one was deprecated and seems the implementation never changed.
On top of it, saw some issues throwing NPE on null deeplinks preventing the onKitCreate for finishing and therefore the configureKits in the core-sdk from initilizing the kit and adding it to the active ones for action forwarding.

BrandonStalnaker reviewed (approved):
LGTM

mauriciolima1988 commented:
Please, support needed [here](https://github.com/mparticle-integrations/mparticle-android-integration-kochava/issues/73)
